### PR TITLE
Refactoring to allow passing both Kafka offset and log event time as LogOffset

### PR DESCRIPTION
**Pull request by poornachandra**
This will allow us to move between Kafka source and HDFS source while querying for logs.

JIRA https://issues.cask.co/browse/CDAP-2053

### CODE REVIEW

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/MockLogReader.java [hunk 1]
```diff
@@ -52,49 +53,51 @@
...
     for (int i = 0; i < MAX; ++i) {
       logMap.put(defaultNamespacedLogBaseDir + "/testApp4/userservice-testService1",
-                 new LogLine(i, "testService1<img>-" + i));
+                 new LogLine(new LogOffset(i, i), "testService1<img>-" + i));
     }
 
     // Add log lines for app testApp2, procedure testProcedure1
     for (int i = 0; i < MAX; ++i) {
       logMap.put(defaultNamespacedLogBaseDir + "/testApp2/procedure-testProcedure1",
-                 new LogLine(i, "testProcedure1<img>-" + i));
```
Comment (sreevatsanraman) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/MockLogReader.java]: nit: procedure is deprecated. you might want to call it service.

**Comment by sreevatsanraman** [comment]
LGTM other than the nit. please fix and merge.

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/MockLogReader.java (hunk 1 above)
Comment (poornachandra) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/log/MockLogReader.java]: We already have a test for service. If procedure is removed I can remove this test, else we can remove it when procedure gets removed.

**Comment by poornachandra** [comment]
Build - http://builds.cask.co/browse/CDAP-DUT1410-3

**Pull request merged**